frontend/src/app/choreography/choreography.component.ts

```diff
@@ -98,29 +103,8 @@ export class ChoreographyComponent implements OnInit {
     }
   }
 
-  generateGrid(): ChoreographyItem[] {
-    return Array(this.choreography.carpet.height * this.choreography.carpet.width)
-      .fill({
-        text: '',
-        color: '',
-        shape: 'rounded',
-        position: ['center', 'center'],
-        sign: null
-      }).map(item => ({
-          ...item,
-          position: [...item.position],
-        })
-      ).map((item, index) => ({
-        ...item,
-        text: index % 3 === 0 ? this.choreography.people[index] : ''
-      }));
-  }
-
-  clearItem(item: ChoreographyItem): void {
-    item.text = '';
-    item.color = '';
-    item.position = ['center', 'center'];
-    item.sign = { text: '', color: '' };
+  get isActiveItemGroup(): boolean {
+    return typeof this.activeChoreographyItem?.text !== 'string' && this.activeChoreographyItem?.text !== null;
```

Comment:
**suggestion:** extract this logic to the same place with `isTwoGroup` and others to make it reusable